qualang_tools/results/data_handler/data_handler.py

```diff
@@ -0,0 +1,112 @@
+        data_folder = Path(data_folder)
+
+    if not data_folder.exists():
+        raise NotADirectoryError(f"Save_data: data_folder {data_folder} does not exist")
+
+    if not isinstance(data, dict):
+        raise TypeError("save_data: 'data' must be a dictionary")
+
+    processed_data = data.copy()
+    for data_processor in data_processors:
+        processed_data = data_processor.process(processed_data)
+
+    json_data = json.dumps(processed_data, indent=4)
+    (data_folder / data_filename).write_text(json_data)
+
+    if metadata is not None:
+        if not isinstance(metadata, dict):
+            raise TypeError("save_data: 'metadata' must be a dictionary")
+
+        with (data_folder / metadata_filename).open("w") as f:
+            json.dump(metadata, f)
+
+    for data_processor in data_processors:
+        data_processor.post_process(data_folder=data_folder)
+
+    return data_folder
+
+
+class DataHandler:
+    default_data_processors = DEFAULT_DATA_PROCESSORS
+    root_data_folder: Path = None
+    folder_pattern: str = DEFAULT_FOLDER_PATTERN
+    data_filename: str = "data.json"
+    metadata_filename: str = "metadata.json"
+
+    def __init__(
+        self,
+        data_processors: Optional[Sequence[DataProcessor]] = None,
+        root_data_folder: Optional[Union[str, Path]] = None,
+        folder_pattern: Optional[str] = None,
+        path: Optional[Path] = None,
+    ):
+        if data_processors is not None:
+            self.data_processors = data_processors
+        else:
+            self.data_processors = [processor() for processor in self.default_data_processors]
+
+        if root_data_folder is not None:
+            self.root_data_folder = root_data_folder
+        if folder_pattern is not None:
+            self.folder_pattern = folder_pattern
+
+        self.path = path
+        self.path_properties = None
+
+    def create_data_folder(
+        self, name: str, idx: Optional[int] = None, use_datetime: Optional[datetime] = None, create: bool = True
+    ) -> Dict[str, Union[str, int]]:
+        """Create a new data folder in the root data folder"""
+        self.path_properties = create_data_folder(
+            root_data_folder=self.root_data_folder,
+            folder_pattern=self.folder_pattern,
+            use_datetime=use_datetime,
+            name=name,
+            idx=idx,
+            create=create,
+        )
+        self.path = self.path_properties["path"]
+        return self.path_properties
+
+    def save_data(self, name, data, metadata=None, idx=None, use_datetime: Optional[datetime] = None):
+        if self.path is None:
+            self.create_data_folder(name, idx=idx, use_datetime=use_datetime)
+
+        return save_data(
+            data_folder=self.path,
+            data=data,
+            metadata=metadata,
+            data_filename=self.data_filename,
+            metadata_filename=self.metadata_filename,
+            data_processors=self.data_processors,
+        )
```

Comment:
Yeah good point. Currently you can do e.g.

```python
data_folder = data_handler.save_data(data)
(data_folder / "configuration.py").save_text(json.dumps(config))
```

But I agree that it would be nice if we could specify default files to be saved

Reply:
@HiroQM I added `DataHandler.additional_files` to autosave additional files. See the description or readme. Are you happy with this solution?

Reply:
Thanks, yes looks great!